### ISSUE TITLE
Docker Port Duplication Issue

### ISSUE THREAD

**Issue by nathanielwendt** [paradoxical-io/docker-client#3]
We are running into issues where our builds/tests are intermittently failing on Jenkins due to port conflicts like the following:
```
com.spotify.docker.client.exceptions.DockerRequestException: Request error: POST unix://localhost:80/containers/161dd11accca3d5512715c9b6788835dc86f7b012e25fb9172b14641bf6e37aa/start: 500, body: {"message":"driver failed programming external connectivity on endpoint wizardly_cori (cf75e5757c817e22e62b09839a71d514ab33e68f0c756c69b313321fe06b9538): Error starting userland proxy: listen tcp 0.0.0.0:38733: bind: address already in use"} 
```

I believe the port sourcing randomization is done here: https://github.com/paradoxical-io/docker-client/blob/master/src/main/java/io/paradoxical/DockerCreator.java#L91.

Is there any way we can change this to operate over a wider range of ports for decreased likelihood in conflicts or perhaps use a mechanism that guarantees selecting an available port?  Let me know your thoughts and I can open a PR!

Thanks!

**Comment by devshorts** [comment]
Hey Nate, if you wanted to make it more robust you could check if the wicket is open by trying to connect to it first, and if it’s already open closing the connection and trying a new port. Also feel free to widen the port range or provide an extension to expose that. If you submit a pr I’ll merge it in quick

**Comment by nathanielwendt** [comment]
Sure, here's my first stab at it: https://github.com/paradoxical-io/docker-client/pull/4

**Comment by nathanielwendt** [comment]
Thanks for the merge, let me know if I can help in getting a release out!

**Comment by devshorts** [comment]
Released in 1.26
